Feature request: pass in custom build command

Looking for some future proofing here.  I know you've been trying to maintain a map of provider to build commands, but a custom command argument would be greatly appreciated for supporting custom builds before you get a chance to update your map in the code.

Right now, I'm trying to build mongo atlas provider 0.8.2:

```
$ m1-terraform-provider-helper install mongodb/mongodbatlas -v v0.8.2
Repo: https://github.com/mongodb/terraform-provider-mongodbatlas
GitRepo: git@github.com:mongodb/terraform-provider-mongodbatlas
2022/01/21 13:17:51 Executingcd /Users/<home_dir>/.m1-terraform-provider-helper && git clone git@github.com:mongodb/terraform-provider-mongodbatlas
Cloning into 'terraform-provider-mongodbatlas'...
Cancel
2022/01/21 13:17:56 Cloning into 'terraform-provider-mongodbatlas'...

HEAD is now at 454ea9c8 Fix 1.2 upgrade/info guide formatting error (#657)
2022/01/21 13:17:56 version: v0.8.2
==> Checking that code complies with gofmt requirements...
gofmt needs running on the following files:
./tools.go
You can use the command: `make fmt` to reformat code.
make: *** [fmtcheck] Error 1
2022/01/21 13:17:59 Bash code did not run successfully: exit status 2
```

So it is another case of `make build` not being sufficient.

Same here!
```
❯ m1-terraform-provider-helper install hashicorp/github -v v3.0.0
Repo: https://github.com/hashicorp/terraform-provider-github
GitRepo: git@github.com:hashicorp/terraform-provider-github
HEAD is now at ec1ed2e v3.0.0
2022/01/24 14:43:09 version: v3.0.0
==> Checking that code complies with gofmt requirements...
gofmt needs running on the following files:
./tools.go
You can use the command: `make fmt` to reformat code.
make: *** [fmtcheck] Error 1
2022/01/24 14:43:10 Bash code did not run successfully: exit status 2

--- a/README.md
+++ b/README.md
@@ -40,6 +40,16 @@ m1-terraform-provider-helper activate # (In case you have not activated the help
 m1-terraform-provider-helper install hashicorp/vault -v v2.10.0 # Install and compile
 ```
 
+### Providing custom build commands
+
+You can override the built-in build command handling by using the `--custom-build-command` flag.
+
+**Explanation**:
+The `install` commands relies on an internal `buildCommands` map to find the correct build command for an provider. For some important providers we have hardcoded different commands, but the default (and fallback) is `make build`. If that does not work for the provider you want to install, you can also pass a custom build command using the `--custom-build-command` flag.
+
+Please refer to the documentation of the provider to find out the build command.
+
+
 ## Development
 
 ### Testing

--- a/cmd/install.go
+++ b/cmd/install.go
@@ -10,6 +10,7 @@ import (
 
 func installCmd() *cobra.Command {
 	var versionString string
+
 	var customBuildCommand string
 
 	cmd := &cobra.Command{

--- a/cmd/install.go
+++ b/cmd/install.go
@@ -10,6 +10,7 @@ import (
 
 func installCmd() *cobra.Command {
 	var versionString string
+	var customBuildCommand string
 
 	cmd := &cobra.Command{
 		Use:   "install [providerName]",
@@ -21,7 +22,7 @@ func installCmd() *cobra.Command {
 			a.Init()
 
 			if a.IsTerraformPluginDirExistent() {
-				a.Install(args[0], versionString)
+				a.Install(args[0], versionString, customBuildCommand)
 			} else {
 				fmt.Fprintln(os.Stdout, "Please activate first")
 			}
@@ -30,6 +31,7 @@ func installCmd() *cobra.Command {
 		},
 	}
 	cmd.Flags().StringVarP(&versionString, "version", "v", "", "The version of the provider")
+	cmd.Flags().StringVar(&customBuildCommand, "custom-build-command", "", "A custom build command to execute instead of the built-in commands")
 
 	return cmd
 }

--- a/internal/app/install.go
+++ b/internal/app/install.go
@@ -208,8 +208,14 @@ func createBuildCommand(providerName string, version string, goPath string) stri
 	return buildCommands["default"][0].command
 }
 
-func (a *App) buildProvider(dir string, providerName string, version string) {
-	buildCommand := createBuildCommand(providerName, version, a.Config.GoPath)
+func (a *App) buildProvider(dir string, providerName string, version string, customBuildCommand string) {
+	var buildCommand string
+	if len(customBuildCommand) > 0 {
+		fmt.Fprintf(os.Stdout, "Using custom build command: \"%s\n\"", customBuildCommand)
+		buildCommand = customBuildCommand
+	} else {
+		buildCommand = createBuildCommand(providerName, version, a.Config.GoPath)
+	}
 	// #nosec G204
 	executeBashCommand(buildCommand, a.Config.ProvidersCacheDir+"/"+dir)
 }
@@ -239,7 +245,7 @@ func (a *App) moveBinaryToCorrectLocation(providerName string, version string, e
 	}
 }
 
-func (a *App) Install(providerName string, version string) bool {
+func (a *App) Install(providerName string, version string, customBuildCommand string) bool {
 	providerData := a.getProviderData(providerName)
 	fmt.Fprintf(os.Stdout, "Repo: %s\n", providerData.Repo)
 
@@ -247,7 +253,7 @@ func (a *App) Install(providerName string, version string) bool {
 	fmt.Fprintf(os.Stdout, "GitRepo: %s\n", gitRepo)
 
 	sourceCodeDir := a.checkoutSourceCode(gitRepo, version)
-	a.buildProvider(sourceCodeDir, providerName, version)
+	a.buildProvider(sourceCodeDir, providerName, version, customBuildCommand)
 
 	name := strings.Split(gitRepo, "/")[1]
 	a.moveBinaryToCorrectLocation(providerName, version, name)

--- a/internal/app/install.go
+++ b/internal/app/install.go
@@ -210,6 +210,7 @@ func createBuildCommand(providerName string, version string, goPath string) stri
 
 func (a *App) buildProvider(dir string, providerName string, version string, customBuildCommand string) {
 	var buildCommand string
+
 	if len(customBuildCommand) > 0 {
 		fmt.Fprintf(os.Stdout, "Using custom build command: \"%s\n\"", customBuildCommand)
 		buildCommand = customBuildCommand

